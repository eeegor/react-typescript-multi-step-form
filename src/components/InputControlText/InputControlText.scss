@import '../../variables';

.input-text,
.input-email {
	display: block;
	width: 100%;
	padding: 0.7rem 0.75rem 0.5rem;
	font-size: 1rem;
	line-height: 2rem;
	color: $color-grey-dark;
	background-clip: padding-box;
	border: 1px solid $color-border;
	border-radius: 0.25rem;
	background-color: white;
	box-shadow: 0 4px 1rem 0 rgba(black, 0.05);
	transition: border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}
