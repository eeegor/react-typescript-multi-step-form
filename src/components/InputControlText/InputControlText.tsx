/**
 * Component InputControlText
 */

import * as React from 'react';
import * as classnames from 'classnames';
import { InputControl, Label, Info } from '..';
import './InputControlText.scss';

export interface InputControlTextProps {
	id: string;
	name: string;
	value?: string;
	label?: string;
	info?: string;
	type?: 'text' | 'email';
	status?: 'success' | 'danger';
	className?: string;
	required?: boolean;
	onChange?: (name: string, event: React.ChangeEvent<HTMLInputElement>) => void;
}

export class InputControlText extends React.Component<InputControlTextProps> {
	render(): JSX.Element {
		const {
			id,
			className,
			required,
			value,
			onChange,
			label,
			type,
			name,
			status,
			info,
		} = this.props;

		return (
			<InputControl id={id} type={type} status={status} className={classnames(className)}>
				<Label className="input-control__label" htmlFor={id || name} label={label} />
				<Info className="input-control__info" label={info} />
				<input
					className={`input-control__input input-${type || 'text'}`}
					id={name}
					name={name}
					type={type}
					onChange={event => onChange && onChange(name, event)}
					value={value || ''}
					autoFocus={true}
					required={required}
				/>
			</InputControl>
		);
	}
}
