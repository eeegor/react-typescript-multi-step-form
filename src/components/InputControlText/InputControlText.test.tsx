import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { InputControlText } from './InputControlText';

describe('InputControlText', () => {
	const callback = jest.fn();
	const testData = {
		id: 'name',
		name: 'name',
		label: 'Your Name',
		value: 'Bob',
		onChange: callback,
	};

	it('renders markup correctly', () => {
		const component = <InputControlText {...testData} />;
		const tree = renderer.create(component).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('change event works', () => {
		const wrapper = mount(<InputControlText {...testData} />);
		wrapper.find('.input-control__input').simulate('change');
		expect(callback).toHaveBeenCalledTimes(1);
	});

	it('should adopt a valid info attribute', () => {
		const info = 'test';
		const wrapper = mount(<InputControlText info={info} {...testData} />);
		expect(
			wrapper
				.find('.input-control__info')
				.first()
				.text()
		).toEqual('test');
	});

	it('should adopt a valid status attribute', () => {
		const status = 'danger';
		const wrapper = mount(<InputControlText status={status} {...testData} />);
		expect(
			wrapper
				.find('.input-control')
				.last()
				.hasClass('input-control--danger')
		).toEqual(true);
	});
});
