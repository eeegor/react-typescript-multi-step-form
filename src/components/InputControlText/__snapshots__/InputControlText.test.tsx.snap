// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputControlText renders markup correctly 1`] = `
<div
  className="input-control input-control-text"
  id="input-control-name"
>
  <label
    className="label input-control__label"
    htmlFor="name"
  >
    Your Name
  </label>
  <input
    autoFocus={true}
    className="input-control__input input-text"
    id="name"
    name="name"
    onChange={[Function]}
    value="Bob"
  />
</div>
`;
